build/Microsoft.DotNet.Cli.BuildInfo.targets

```diff
@@ -0,0 +1,40 @@
+<Project ToolsVersion="15.0">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.tasks))\dir.tasks" />
+
+  <Target Name="WriteBuildInfoProps"
+          Condition=" !Exists('$(BuildInfoProps)') "
```

Comment:
How about I always write the file? That can't be too bad perf-wise but will make the code much simpler.